lms/djangoapps/discussion_api/api.py

```diff
@@ -487,7 +488,7 @@ def _check_editable_fields(cc_content, data, context):
     )
 
 
-def _do_extra_actions(api_content, cc_content, request_fields, actions_form, context):
+def _do_extra_actions(api_content, cc_content, request_fields, actions_form, context, request):
```

Comment:
This function is getting unweildy.  Consider extracting methods from it that describe what it does. Something like:

    _handle_following_field 
    _handle_abuse_flagged_field
    _handle_voted_field

Also, the `else: assert field == 'voted'` bit is weird.  That should probably be an `elif field == 'voted'` with a new `else` block that raises a more appropriate exception (maybe a KeyError).